rollup.config.ts

```diff
@@ -15,7 +15,10 @@ const config: Array<RollupWatchOptions> = [
   },
   {
     input: "./.build/src/index.d.ts",
-    output: [{ file: pkg.types, format: "es" }],
+    output: [
+      { file: pkg.exports.types.import, format: "es" },
+      { file: pkg.exports.types.require, format: "commonjs", exports: "named" },
```

Comment:
this won’t really have any effect, as we override these output options; and exclusively only generate ES exports.

Reply:
We need to generate both a `.d.mts` and a `.d.cts` file. These two files would be identical but they need to be different files.

I'll leave it to you to figure what the best way to do that is. Whether it be to use this PR as is, use `format: "es"` for both outputs or just generate one then copy-paste it.

Reply:
Could even just shorten it to:

```ts
output: [{ file: pkg.exports.types.import }, { file: pkg.exports.types.require }]
```

Reply:
I think the rollup typings require you to define at least one `format`. I can take a look on the weekend again, I was distracted with other matters these days.